From 7de61d0e3230f2d12ca9ba4cca941c7a2bab55f8 Mon Sep 17 00:00:00 2001
From: Vasu Kantubukta <vasu.kantubukta@lge.com>
Date: Fri, 23 Feb 2018 19:39:07 +0530
Subject: [PATCH] Fix for connection lost issue for setipv6 API

:Release Notes:
Fix for connection lost issue for setipv6 API

:Detailed Notes:
Issue was happening due to clearing the IPv4 network config.
This patch fix the issue by validating the type check for ipconfig
before enabling the ipv6 network

Upstream-Status : Pending
---
 src/service.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/service.c b/src/service.c
index 84ab2e3..41a5235 100644
--- a/src/service.c
+++ b/src/service.c
@@ -3499,9 +3499,11 @@ static DBusMessage *set_property(DBusConnection *conn,
 			ipv6_configuration_changed(service);
 
 		if (is_connecting(service) || is_connected(service)) {
-			__connman_network_enable_ipconfig(service->network,
+			if(type==CONNMAN_IPCONFIG_TYPE_IPV4)
+				__connman_network_enable_ipconfig(service->network,
 							service->ipconfig_ipv4);
-			__connman_network_enable_ipconfig(service->network,
+			else
+				__connman_network_enable_ipconfig(service->network,
 							service->ipconfig_ipv6);
 		}
 
-- 
1.9.1

